### PR TITLE
mgr/nfs: clarify in the output message

### PR DESCRIPTION
**Pull request by ajarr**
This also fixes the misleading output message that said NFS cluster
    creation or removal was complete when the `nfs cluster create/rm` command
    returned. The `nfs cluster create/rm` command triggers the
    asynchronous creation/removal of the NFS daemons and ingress
    service of the cluster.
    
    Fixes: https://tracker.ceph.com/issues/55299
    Signed-off-by: Ramana Raja <rraja@redhat.com>

doc/mgr/nfs: Add commands to check the statuses
    
    .. of NFS and ingress services after creating/deleting a NFS cluster.
    The `nfs cluster info` command is not sufficient to show that the
    NFS cluster is created/deleted as expected.
    
    Signed-off-by: Ramana Raja <rraja@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/mgr/nfs.rst [hunk 1]
```diff
@@ -128,6 +138,17 @@ Delete NFS Ganesha Cluster
 
 This deletes the deployed cluster.
 
+Since the removal of NFS daemon(s) and the ingress service (if deployed) is
+asynchronous, check the statuses of the NFS and ingress services to make sure
+that they have been deleted successfully. If your orchestrator is cephadm,
+execute the following and check whether no services are reported:
```
Comment (anthonyeleven) [inline doc/mgr/nfs.rst:144]: Suggest "execute the following to check that the services are no longer reported."

Comment (ajarr) [reply doc/mgr/nfs.rst]: Done! Thanks!

**Comment by anthonyeleven** [comment]
Thanks!  I've found that in technical documentation subtle wording changes can make all the difference for user interpretation.

LGTM from a doc perspective.  Probably best to get a thumbs-up from someone else for the code part.

**Review by anthonyeleven — APPROVED** [review]

**Review by adk3798 — APPROVED** [review]
LGTM

**Comment by ajarr** [comment]
Let me add a mgr/nfs tracker ticket to help keep track of backports.

File: src/pybind/mgr/nfs/cluster.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def create_nfs_cluster(
 
             if cluster_id not in available_clusters(self.mgr):
                 self._call_orch_apply_nfs(cluster_id, placement, virtual_ip, port)
-                return 0, "NFS Cluster Created Successfully", ""
+                return 0, "Creating NFS cluster", ""
```
Comment (phlogistonjohn) [inline src/pybind/mgr/nfs/cluster.py:124]: I'm approve of this change, but in general I would encourage traditional unixy practices of not printing stuff out (unless asked) on success. I had to work around the fact that the nfs module does print these "friendly" messages in go-ceph (see: https://github.com/ceph/go-ceph/blob/master/common/admin/nfs/export.go#L141 for an example).
And if we must print stuff out we should be consistent and print structured responses that contain an id/key that could be fed back into a status command.

**Review by phlogistonjohn — APPROVED** [review]

File: src/pybind/mgr/nfs/cluster.py (hunk 2 above)
Comment (ajarr) [reply src/pybind/mgr/nfs/cluster.py:124]: Good points. Out of curiosity, I checked the output of `fs volume create/rm` commands that creates/deletes a CephFS file system and creates/removes MDS daemons asynchronously.

`fs volume create` prints nothing on success
`fs volume rm` prints a message "metadata pool: cephfs.a.meta data pool: ['cephfs.a.data'] removed" on success

We're not consistent with output messages in `fs volume create/rm` commands as well.

I like the idea of not printing anything when the commands `nfs cluster create/rm` succeed. What do you think?

Comment (ajarr) [reply src/pybind/mgr/nfs/cluster.py]: > Let me add a mgr/nfs tracker ticket to help keep track of backports.

Done.

**Comment by ajarr** [comment]
@phlogistonjohn PTAL.  I made the change to not output any message when `nfs cluster create/rm` commands succeeds.

**Comment by ajarr** [comment]
@alfonsomthd would changing the stdout of successul `nfs cluster create/rm` command execution to empty string affect dashboard code?

File: doc/mgr/nfs.rst [hunk 3]
```diff
@@ -76,6 +76,16 @@ service.
 For more details, refer :ref:`orchestrator-cli-placement-spec` but keep
 in mind that specifying the placement via a YAML file is not supported.
 
+Since the deployment of NFS daemon(s) and the ingress service is asynchronous,
+check the statuses of the NFS and ingress services to make sure that the
+services are running. If your orchestrator is cephadm, then check using:
```
Comment (anthonyeleven) [inline doc/mgr/nfs.rst:81]: Suggest reducing duplication, the "don't use  a word twice in a sentence" factor.  Something like:

Deployment of NFS daemons and the ingress service is asynchronous: the command may return before the services have completely started.  You may wish to check that these services do successfully start and stay running.  When using `cephadm` orchestration, these commands check service status:

**Comment by alfonsomthd** [comment]
> @alfonsomthd would changing the stdout of successul `nfs cluster create/rm` command execution to empty string affect dashboard code?

@ajarr I don't think so. @avanthakkar @nizamial09 as you're more updated than me, can you take a look, please?

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Comment by nizamial09** [comment]
> Basically to confirm that cluster creation/deletion through the service form still works.

@ajarr @alfonsomthd testing this locally on the dashboard works without any issues.

**Review by phlogistonjohn — APPROVED** [review]
Code change LGTM, thanks.

**Comment by ajarr** [comment]
Just made doc changes as suggested in https://github.com/ceph/ceph/pull/45843#discussion_r853582615 .
No code changes were made.

**Pull request merged**